### PR TITLE
handle-errors

### PR DESCRIPTION
**Pull request by bean00**
Handling generic errors is done.

I've replaced `request` with my error handler in my tests, and then just call the error handler in the `catch` clause of the `fetch` call.

### CODE REVIEW

**Comment by tbaik** [comment]
To add to Jason's comment - you'd use sinon if you assert on the fact that your mock has been called on. In this case, it seems like you're really just trying to test that you show an error

File: src/App.vue [hunk 1]
```diff
@@ -24,7 +24,8 @@
       url: "",
       body: "",
       language: "",
-      response: undefined
+      response: undefined,
+      wasResponseSuccessful: false
```
Comment (jdesrosiers) [inline src/App.vue]: Is this flag necessary? Does `wasResponseSuccessful` mean anything different than `response === undefined`?

File: src/App.vue [hunk 2]
```diff
@@ -41,7 +42,11 @@
       },
       request(url) {
         fetch(url, { headers: { Accept: "application/json" } })
-          .then((response) => this.response = response);
+          .then((response) => this.response = response)
+          .catch(function(error) {
```
Comment (jdesrosiers) [inline src/App.vue]: You should prefer the `=>` notation for functions unless you have a good reason not to.

File: src/App.vue (hunk 1 above)
Comment (bean00) [reply src/App.vue]: Removed

File: src/App.vue (hunk 2 above)
Comment (bean00) [reply src/App.vue]: Done

File: src/components/UrlBar.vue [hunk 3]
```diff
@@ -5,26 +5,65 @@
...
+       @keyup.enter="onEnter"
        aria-label="URL"
        aria-describedby="basic-addon2">
   </div>
 </template>
+       <!--:class="{ invalid: !isUrlValid }"-->
+       <!--@keydown="onKeydown"-->
 
 <script>
+  //import Ajv from "ajv";
```
Comment (jdesrosiers) [inline src/components/UrlBar.vue]: This can't be merged with all this commented out code. Do some cleanup first.

**Pull request merged**